source/common/http/http2/codec_impl.h

```diff
@@ -84,6 +84,17 @@ class ConnectionImpl : public virtual Connection, Logger::Loggable<Logger::Id::h
   Protocol protocol() override { return Protocol::Http2; }
   void shutdownNotice() override;
   bool wantsToWrite() override { return nghttp2_session_want_write(session_); }
+  // Propogate network connection watermark events to each stream on the connection.
+  void onUnderlyingConnectionAboveWriteBufferHighWatermark() override {
```

Comment:
I could go either way. On one hand I agree, on the other hand, the other approach would put logic closer to where it is actually needed. I would just leave it for now if you are happy with it.